### PR TITLE
Blanket order

### LINKED ISSUE #13733: Rate and Quantity Contract / Blanket Order
Fixed price + max qty contracted, purchase orders will be released periodically against a rate contract

### 
- [x] Selling or Buying
- [x] Customer and Supplier + Billing Address
- [x] From / To Date
- [x] Item table (item, forcasted qty and rate[mandatory], order qty)
- [x] terms and conditions

###

create sales and purchase order from the blanket order

### PR DESCRIPTION
**Pull request by manassolanki**
Fixes https://github.com/frappe/erpnext/issues/13733

**Blanket Order and order creation from it**
blanket order 1

**Auto selection of Blanket Order's rate in the Sales Order**
blanket order 2

### CODE REVIEW

**Review by nabinhait — CHANGES_REQUESTED** [review]
- [x] Add tests
- [x] Add docs

**Comment by nabinhait** [comment]
- [ ] Add screenshots / GIF

**Comment by manassolanki** [comment]
@nabinhait Done

**Review by nabinhait — APPROVED** [review]

File: erpnext/controllers/stock_controller.py [hunk 1]
```diff
@@ -342,6 +342,18 @@ def validate_inspection(self):
 				if self.docstatus==1:
 					raise frappe.ValidationError
 
+	def update_blanket_order(self, cancel=False):
+		for item in self.items:
+			if item.blanket_order:
+				ordered_quantity, doc_name = frappe.db.get_value("Blanket Order Item", {"parent": item.blanket_order}, ["ordered_quantity", "name"])
+				if not cancel:
+					ordered_quantity = ordered_quantity + item.qty
```
Comment (nabinhait) [inline erpnext/controllers/stock_controller.py]: Write this type function using `select sum(qty)` query, this helps to correct if goes wrong. For example, you forgot to call this function on STOP and UNSTOP events.

**Review by nabinhait — CHANGES_REQUESTED** [review]

**Comment by nabinhait** [comment]
merged from console

**Pull request closed without merging**